sql/scan.go

```diff
@@ -682,33 +683,80 @@ func (v *qnameVisitor) Visit(expr parser.Expr, pre bool) (parser.Visitor, parser
 	if !pre || v.err != nil {
 		return nil, expr
 	}
-	qname, ok := expr.(*parser.QualifiedName)
-	if !ok {
-		return v, expr
-	}
 
-	v.err = qname.NormalizeColumnName()
-	if v.err != nil {
-		return v, expr
-	}
-	if qname.IsStar() {
+	switch t := expr.(type) {
+	case *parser.QualifiedName:
+		qname := t
+
+		v.err = qname.NormalizeColumnName()
+		if v.err != nil {
+			return nil, expr
+		}
+		if qname.IsStar() {
+			v.err = fmt.Errorf("qualified name \"%s\" not found", qname)
+			return nil, expr
+		}
+
+		desc := v.getDesc(qname)
+		if desc != nil {
+			name := qname.Column()
+			for _, col := range v.visibleCols {
+				if !equalName(name, col.Name) {
+					continue
+				}
+				return v, v.getQVal(col)
+			}
+		}
+
 		v.err = fmt.Errorf("qualified name \"%s\" not found", qname)
 		return nil, expr
-	}
 
-	desc := v.getDesc(qname)
-	if desc != nil {
-		name := qname.Column()
-		for _, col := range v.visibleCols {
-			if !equalName(name, col.Name) {
-				continue
+	case *parser.FuncExpr:
+		// Special case handling for COUNT(*) and COUNT(foo.*), expanding the star
```

Comment:
The descriptor is already loaded. Changing the downstream machinery to be able to deal with unexpanded identifiers doesn't seem feasible. Essentially, we're transforming a query like:

```
SELECT COUNT(*) FROM t
```

into

```
COUNT(SELECT k FROM t)
```

The `*` absolutely needs to be transformed into some set of columns (perhaps a single column). This is similar to the way `*` is expanded in `scanNode.initTargets`. Could perhaps try to unify the two expansions, but as you've noted we only need to expand `*` here to a non-NULL column.